src/vm/syncblk.inl

```diff
@@ -178,7 +178,7 @@ FORCEINLINE AwareLock::LeaveHelperAction AwareLock::LeaveHelper(Thread* pCurThre
     _ASSERTE((size_t)m_MonitorHeld & 1);
     _ASSERTE(m_Recursion >= 1);
 
-#if defined(_DEBUG) && defined(TRACK_SYNC)
+#if defined(_DEBUG) && defined(TRACK_SYNC) && !defined(CROSSGEN_COMPILE)
```

Comment:
I think that if you defined TRACK_SYNC in both clrdefinitions.cmake and crossgen.cmake, you would not need this added condition.